tcn_hpl/data/utils/bbn.py

```diff
@@ -0,0 +1,292 @@
+#!/usr/bin/env python3
+
+import dataclasses
+import os
+from pathlib import Path
+import re
+import typing
+import warnings
+
+import click
+import cv2
+import kwcoco
+import numpy as np
+import numpy.typing as npt
+from tqdm import tqdm
+import yaml
+
+
+# Regex to match a BBN Truth file and parse out the "basename" for which we
+# should find a matching `.mp4` file next to it.
+RE_TRUTH_FILENAME = re.compile(r"^(?P<basename>.*)\.skill_labels_by_frame\.txt$")
+
+# Parsing a BBN Truth file line into component parts.
+# Assumes that surrounding whitespace has been stripped.
+RE_BBN_TRUTH_LINE = re.compile(
+    r"^(?P<start_frame>\d+)\s+(?P<end_frame>\d+)\s+(?P<task_name>[\w\d]+)\s+"
+    r"(?:Error: (?P<error>.*) S\s+)?(?P<description>.*)$"
+)
+
+
+@dataclasses.dataclass
+class VideoInfo:
+    truth_path: Path
+    mp4_path: Path
+    frames_dir: Path = dataclasses.field(init=False)
```

Comment:
This is a very picky review item: do we have a differentiation between `_path` and `_dir`?